src/m_cloudbox.cc

```diff
@@ -578,6 +579,452 @@ void cloudboxSetManuallyAltitude(  // WS Output:
+                           const Vector& p_grid,
+                           const Vector& lat_grid,
+                           const Vector& lon_grid,
+                           const Tensor3& z_field,
+                           const Matrix& z_surface,
+                           const Index& stokes_dim,
+                           const Vector& scat_za_grid,
+                           const Vector& scat_aa_grid,
+                           const Vector& f_grid,
+                           //const Index&    p_interp_order,
+                           const Index& za_interp_order,
+                           const Index& za_restrict,
+                           const Index& cos_za_interp,
+                           const Numeric& za_extpolfac,
+                           const Index& aa_interp_order,
+                           const Verbosity&) {
+
+  //--- Check input -----------------------------------------------------------
+  if (!(atmosphere_dim == 1 || atmosphere_dim == 3))
+    throw runtime_error("The atmospheric dimensionality must be 1 or 3.");
+  if (jacobian_do)
+    throw runtime_error(
+        "This method does not support jacobians (jacobian_do must be 0)");
+  if (!cloudbox_on)
+    throw runtime_error(
+        "The cloud box is not activated and no outgoing "
+        "field can be returned.");
+  if (cloudbox_limits.nelem() != 2 * atmosphere_dim)
+    throw runtime_error(
+        "*cloudbox_limits* is a vector which contains the upper and lower\n"
+        "limit of the cloud for all atmospheric dimensions.\n"
+        "So its length must be 2 x *atmosphere_dim*");
+  if (scat_za_grid.nelem() == 0)
+    throw runtime_error(
+        "The variable *scat_za_grid* is empty. Are dummy "
+        "values from *cloudboxOff used?");
+  if (!(za_interp_order < scat_za_grid.nelem()))
+    throw runtime_error(
+        "Zenith angle interpolation order *za_interp_order*"
+        " must be smaller\n"
+        "than number of angles in *scat_za_grid*.");
+  if (atmosphere_dim > 1 && !(aa_interp_order < scat_aa_grid.nelem()))
+    throw runtime_error(
+        "Azimuth angle interpolation order *aa_interp_order*"
+        " must be smaller\n"
+        "than number of angles in *scat_aa_grid*.");
+  if (doit_i_field.nlibraries() != f_grid.nelem())
+    throw runtime_error(
+        "Inconsistency in size between f_grid and doit_i_field! "
+        "(This method does not yet handle dispersion type calculations.)");
+  //---------------------------------------------------------------------------
+
+  // At each lat/lon, one value of the intensity field is defined at the
+  // surface. Simplest way to handle this is to insert z_surface into z_field
+  Tensor3 z_with_surface = z_field;
+  for (Index ilat=0; ilat < z_surface.nrows(); ilat++) {
+    for (Index ilon=0; ilon < z_surface.ncols(); ilon++) {
+      Index ip = 0;
+      for (; z_surface(ilat,ilon) >= z_field(ip+1,ilat,ilon); ip++) {}
```

Comment:
This for loop with an empty body gives the impression that it was accidentally left empty.
A while loop would make your intention clearer:

```cpp
while (z_surface(ilat,ilon) >= z_field(ip+1,ilat,ilon)) {
  ip++;
}